stl/inc/filesystem

```diff
@@ -275,8 +275,8 @@ namespace filesystem {
 #ifdef __cpp_char8_t
     template <class _Conversion>
     _NODISCARD wstring _Convert_stringoid_to_wide(const basic_string_view<char8_t> _Input, _Conversion) {
-        static_assert(
-            is_same_v<_Conversion, _Normal_conversion>, "invalid value_type, see N4810 D.17 [depr.fs.path.factory]/1");
+        static_assert(_Is_any_of_v<_Conversion, _Normal_conversion, _Utf8_conversion>);

```

Comment:
should we preserve the nice error message?

Reply:
`_Normal_conversion` and `_Utf8_conversion` are the only two types that could ever be passed here; the `static_assert` is now a pure paranoia check to avoid a future maintainer doing something silly (similar to 259). It may make sense to change both to `_STL_INTERNAL_STATIC_ASSERT` to make it clear that it should be impossible for a user to do something that results in misuse of these internal functions. (We didn't have `_STL_INTERNAL_STATIC_ASSERT` in our toolbox when this line or 259 were written.) I think I'll make that change.